### PR TITLE
Indentify and fetch data

### PR DESCRIPTION
**Pull request by georgiana-b**
Solved issues: 
-  An ID data script, which builds publishers and sources lists
-  A fetch script, which gets the sources to a local path
  -  Do away with the whole local server thing that was added here, instead, when fetching, add another column to the sources.csv that declares a local path (call the column cache)
  -  Save the fetched files in fetched/\* so we have an audit trail of data assessed via git commits. Ensure that the fetch script clears the fetched directory completely before repopulating, so we can also see files that disappear over time.
-  Fix the scripts that identify assessable data, and further manually curate the identified data (if it helps streamline the scripts) so that we only assess the relevant publishers, and only on their 25k data (not other documents of spend data)
- Identify and exclude from sources HTML files.

### CODE REVIEW

File: scripts/id_data.py [hunk 1]
```diff
@@ -83,34 +85,56 @@ def get_organization_data(organization):
...
         groups = organization.get('groups')
-        parent = ''
+        parent = [group.get('name')]
         for group in groups:
             if parent and group.get('name'):
                 parent += ' / ' + group.get('name')
             elif group.get('name'):
                 parent += group.get('name')
         publisher['parent_id'] = parent
-    if publisher['id']:
```
Comment (pwalsh) [inline scripts/id_data.py:109]: Why remove this?

File: scripts/make_local_sources.py [hunk 2]
```diff
@@ -2,7 +2,10 @@
 """This script makes a new CSV sources file with local urls.
```
Comment (pwalsh) [inline scripts/make_local_sources.py]: Why do we still need this?

File: scripts/id_data.py [hunk 3]
```diff
@@ -83,34 +85,56 @@ def get_organization_data(organization):
...
     # Get the data.
-    publisher = {'id': '', 'title': '', 'type': '', 'contact': '', 'email': '', 'parent_id': '', 'homepage': ''}
-    if organization.get('name'):
-        publisher['id'] = organization.get('name')
-    if organization.get('title'):
-        publisher['title'] = organization.get('title')
+    publisher = {}
+    publisher['title'] = organization.get('title', '')
+    publisher['id'] = organization.get('name', '')
+    if publisher['id']:
```
Comment (georgiana-b) [inline scripts/id_data.py:95]: It's here.

Comment (pwalsh) [inline scripts/id_data.py:95]: ok.

**Comment by pwalsh** [comment]
@georgiana-b the last commit to exclude html files still has issues:
1. It does not take into account web pages that are HTML but without a file extension. So we have false positives in `sources.csv` like here https://github.com/okfn/data-quality-uk-25k-spend/blob/feature/refactor/data/sources.csv#L19
2. It looks to me like you are still just removing all HTML (apart from above bug), when what we said was remove HTML with status of 200. Why? Because if we get a 404, we can assume that a CSV file URL has died, and that should be part of our reporting (talked about this yesterday)

**Pull request merged**